internal/pkg/agent/cmd/run.go

```diff
@@ -287,6 +295,27 @@ func runElasticAgent(ctx context.Context, cancel context.CancelFunc, override cf
 	}
 	defer control.Stop()
 
+	// create symlink from /run/elastic-agent.sock to `paths.ControlSocket()` when running as root
+	// this provides backwards compatibility as the control socket was moved with the addition of --unprivileged
+	// option during installation
+	if isRoot && paths.RunningInstalled() && paths.ControlSocketRunSymlink != "" {
```

Comment:
Shouldn't there be an explicit platform check here, or is it implicit in `ControlSocketRunSymlink` being the empty string? If that is the case I would add a comment explaining that this isn't necessary on Windows since that isn't obvious without the context of reading this entire at once.

Reply:
Yes on Windows the `paths.ControlSocketRunSymlink` is `""`. I will add a comment.

Reply:
Done.